src/System.CodeDom/tests/System/CodeDom/Compiler/ExecutorTests.cs

```diff
@@ -0,0 +1,100 @@
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.Linq;
+using Xunit;
+
+namespace System.CodeDom.Compiler.Tests
+{
+    public class ExecutorTests : FileCleanupTestBase
+    {
+        private static readonly string s_cmd = PlatformDetection.IsWindows ? "ipconfig" : "ifconfig"; // arbitrary commands to validate output
+
+        [SkipOnTargetFramework(~TargetFrameworkMonikers.Netcoreapp, "Not supported on .NET Core")]
+        [Fact]
+        public void Impersonation_Unsupported_Throws()
+        {
+            string outputName = null, errorName = null;
+            Assert.Throws<PlatformNotSupportedException>(() => Executor.ExecWaitWithCapture((IntPtr)1, null, null, ref outputName, ref errorName));
+            Assert.Throws<PlatformNotSupportedException>(() => Executor.ExecWaitWithCapture((IntPtr)1, null, null, null, ref outputName, ref errorName));
+        }
+
+        [Fact]
+        public void NullTempFileCollection_Required_Throws()
+        {
+            string outputName = null, errorName = null;
+            Assert.Throws<NullReferenceException>(() => Executor.ExecWaitWithCapture("", null, ref outputName, ref errorName));
+        }
+
+        [Fact]
+        public void ExecWait_OutputCaptured()
+        {
+            var tfc = new TempFileCollection(TestDirectory);
+
+            Executor.ExecWait(s_cmd, tfc);
+            Assert.Equal(3, tfc.Count);
+            Assert.NotEmpty(File.ReadAllText(tfc.BasePath + ".out"));
+            Assert.Empty(File.ReadAllText(tfc.BasePath + ".err"));
+        }
+
+        [Fact]
+        public void ExecWaitWithCapture_NullNames_OutputCaptured()
+        {
+            var tfc = new TempFileCollection(TestDirectory);
+
+            string outputName = null;
+            string errorName = null;
+
+            Executor.ExecWaitWithCapture(s_cmd, tfc, ref outputName, ref errorName);
+
+            Assert.Equal(3, tfc.Count);
+
+            Assert.NotNull(outputName);
+            Assert.Contains(outputName, tfc.Cast<string>());
+
+            Assert.NotNull(errorName);
+            Assert.Contains(errorName, tfc.Cast<string>());
+
+            Assert.NotEmpty(File.ReadAllText(outputName));
+            Assert.Empty(File.ReadAllText(errorName));
+        }
+
+        [Theory]
+        [InlineData(false, false)]
+        [InlineData(false, true)]
+        [InlineData(true, false)]
+        [InlineData(true, true)]
+        public void ExecWaitWithCapture_FilesSpecified_OutputCaptured(bool specifyOutput, bool specifyError)
+        {
+            var tfc = new TempFileCollection(TestDirectory);
+
+            string outputName = specifyOutput ? GetTestFilePath() : null;
+            string errorName = specifyError ? GetTestFilePath() : null;
+
+            Executor.ExecWaitWithCapture(s_cmd, tfc, ref outputName, ref errorName);
+
+            Assert.Equal(
```

Comment:
nit: This assert is a bit difficult to reason about. You already have asserts within `!specifyOutput` and `!specifyError` blocks. Add the four branches and asserts there.

Reply:
I simplified.